sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosResponseDiagnosticsTest.java

```diff
@@ -0,0 +1,149 @@
+            fail("request should fail as partition key is wrong");
+        } catch (CosmosClientException exception) {
+            String diagnostics = exception.getCosmosResponseDiagnostics().toString();
+            assertThat(exception.getStatusCode()).isEqualTo(HttpConstants.StatusCodes.NOTFOUND);
+            assertThat(diagnostics).contains("Connection Mode : " + ConnectionMode.GATEWAY);
+            assertThat(diagnostics).contains("Gateway statistics");
+            assertThat(diagnostics).contains("Status Code : 404");
+            assertThat(diagnostics).contains("Operation Type : " + OperationType.Read);
+            assertThat(exception.getCosmosResponseDiagnostics().getRequestLatency()).isNotNull();
+            System.out.println(diagnostics);
+        }
+    }
+
+    @Test(groups = {"simple"})
+    public void testSystemDiagnosticsForSystemStateInformation() throws CosmosClientException {
+        CosmosItemProperties cosmosItemProperties = new CosmosItemProperties();
+        cosmosItemProperties.setId(UUID.randomUUID().toString());
+        CosmosItemResponse createResponse = this.cosmosContainer.createItem(cosmosItemProperties);
+        String diagnostics = createResponse.getCosmosResponseDiagnostics().toString();
+        assertThat(diagnostics).contains("System State Information ------");
+        assertThat(diagnostics).contains("Used Memory :");
+        assertThat(diagnostics).contains("Available Memory :");
+        assertThat(diagnostics).contains("CPU Process Load :");
+        assertThat(diagnostics).contains("CPU System Load :");
+
+        assertThat(createResponse.getCosmosResponseDiagnostics().getRequestLatency()).isNotNull();
+    }
+
+    @Test(groups = {"simple"})
+    public void testDirectDiagnostics() {
+        ConnectionPolicy connectionPolicy = new ConnectionPolicy();
+        connectionPolicy.setConnectionMode(ConnectionMode.DIRECT);
+        CosmosClient directClient = clientBuilder().setConnectionPolicy(connectionPolicy).buildClient();
+        CosmosContainer cosmosContainer = directClient.getDatabase(DATABASE_NAME).getContainer(CONTAINER_NAME);
+        CosmosItemProperties cosmosItemProperties = new CosmosItemProperties();
+        cosmosItemProperties.setId(UUID.randomUUID().toString());
+        try {
+            CosmosItemResponse createResponse = cosmosContainer.createItem(cosmosItemProperties);
+            String diagnostics = createResponse.getCosmosResponseDiagnostics().toString();
+            assertThat(diagnostics).contains("Connection Mode : " + ConnectionMode.DIRECT);
+            assertThat(diagnostics).contains("StoreResponseStatistics");
+            assertThat(diagnostics).doesNotContain("Gateway request URI :");
+            assertThat(diagnostics).contains("AddressResolutionStatistics");
+            assertThat(createResponse.getCosmosResponseDiagnostics().getRequestLatency()).isNotNull();
+        } catch (CosmosClientException e) {
+            fail(e.getMessage());
```

Comment:
don't catch and fail, if this the test method throws, testng will automatically fail. drop try/catch , here too and anywhere else applicable

Reply:
Done